drivers/SmartThings/zwave-lock/src/schlage-lock/init.lua

```diff
@@ -54,26 +54,35 @@ local function reload_all_codes(self, device, cmd)
 end
 
 local function set_code(self, device, cmd)
-  -- it's copied function from defaults with additional check for Schlage's configuration
-  if (cmd.args.codeName ~= nil) then
-    if (device:get_field(constants.CODE_STATE) == nil) then device:set_field(constants.CODE_STATE, {}) end
-    local code_state = device:get_field(constants.CODE_STATE)
-    code_state["setName"..cmd.args.codeSlot] = cmd.args.codeName
-    device:set_field(constants.CODE_STATE, code_state)
-  end
-  local send_set_user_code = function ()
-    device:send(UserCode:Set({
-      user_identifier = cmd.args.codeSlot,
-      user_code = cmd.args.codePIN,
-      user_id_status = UserCode.user_id_status.ENABLED_GRANT_ACCESS})
-    )
-  end
-  local current_code_length = device:get_latest_state("main", capabilities.lockCodes.ID, capabilities.lockCodes.codeLength.NAME)
-  if current_code_length ~= nil then
-    device:send(Configuration:Get({parameter_number = SCHLAGE_LOCK_CODE_LENGTH_PARAM.number}))
-    device.thread:call_with_delay(DEFAULT_COMMANDS_DELAY, send_set_user_code)
+  if (cmd.args.codePIN == "") then
+    self:inject_capability_command(device, {
+      capability = capabilities.lockCodes.ID,
+      command = capabilities.lockCodes.commands.nameSlot.NAME,
+      args = cmd.args,
+      positional_args = cmd.positional_args
+    })
   else
-    send_set_user_code()
+    -- copied from defaults with additional check for Schlage's configuration
+    if (cmd.args.codeName ~= nil and cmd.args.codeName ~= "") then
+      if (device:get_field(constants.CODE_STATE) == nil) then device:set_field(constants.CODE_STATE, {}) end
```

Comment:
this file also needs the `lock_constants` -> `lock_utils` change

Reply:
This is in the zwave lock which I haven't touched yet.  The only reason changes are showing up here is that I based this off your PR, so it included the rename fix.

Reply:
I should probably try to fix the commits though so this isn't included in a commit for "Zigbee lock code fixes"